server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSPerformSPOptionsCheckUtility.java

```diff
@@ -0,0 +1,282 @@
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+package org.eclipse.lsp.cobol.implicitDialects.cics.utility;
+
+        import org.antlr.v4.runtime.ParserRuleContext;
+
+        import org.eclipse.lsp.cobol.common.dialects.DialectProcessingContext;
+        import org.eclipse.lsp.cobol.common.error.ErrorSeverity;
+        import org.eclipse.lsp.cobol.common.error.SyntaxError;
+        import org.eclipse.lsp.cobol.implicitDialects.cics.CICSLexer;
+        import org.eclipse.lsp.cobol.implicitDialects.cics.CICSParser;
+
+        import java.util.HashMap;
+        import java.util.List;
+        import java.util.Map;
+
+        import static org.eclipse.lsp.cobol.implicitDialects.cics.CICSParser.*;
+
+/** Checks CICS Create System Command rules for required and invalid options */
+public class CICSPerformSPOptionsCheckUtility extends CICSOptionsCheckBaseUtility {
+
+    public static final int RULE_INDEX = RULE_cics_perform;
+    private static final Map<Integer, ErrorSeverity> DUPLICATE_CHECK_OPTIONS =
+            new HashMap<Integer, ErrorSeverity>() {
+                {
+                    put(CICSLexer.DELETSHIPPED, ErrorSeverity.WARNING);
+                    put(CICSLexer.DUMPCODE, ErrorSeverity.ERROR);
+                    put(CICSLexer.DUMP, ErrorSeverity.WARNING);
+                    put(CICSLexer.TITLE, ErrorSeverity.ERROR);
+                    put(CICSLexer.TITLELENGTH, ErrorSeverity.ERROR);
+                    put(CICSLexer.CALLER, ErrorSeverity.ERROR);
+                    put(CICSLexer.CALLERLENGTH, ErrorSeverity.ERROR);
+                    put(CICSLexer.ENDAFFINITY, ErrorSeverity.WARNING);
+                    put(CICSLexer.NETNAME, ErrorSeverity.ERROR);
+                    put(CICSLexer.NETID, ErrorSeverity.ERROR);
+                    put(CICSLexer.JVMSERVER, ErrorSeverity.ERROR);
+                    put(CICSLexer.JVMTYPE, ErrorSeverity.ERROR);
+                    put(CICSLexer.JVM, ErrorSeverity.WARNING);
+                    put(CICSLexer.DUMPTYPE, ErrorSeverity.ERROR);
+                    put(CICSLexer.ALL, ErrorSeverity.WARNING);
+                    put(CICSLexer.JAVACORE, ErrorSeverity.WARNING);
+                    put(CICSLexer.HEAP, ErrorSeverity.WARNING);
+                    put(CICSLexer.SNAPTRACE, ErrorSeverity.WARNING);
+                    put(CICSLexer.GATHER, ErrorSeverity.WARNING);
+                    put(CICSLexer.STACKTRACE, ErrorSeverity.ERROR);
+                    put(CICSLexer.GATHERTYPE, ErrorSeverity.ERROR);
+                    put(CICSLexer.DIAGNOSTICS, ErrorSeverity.WARNING);
+                    put(CICSLexer.TASKID, ErrorSeverity.ERROR);
+                    put(CICSLexer.LIBERTY, ErrorSeverity.WARNING);
```

Comment:
error.

re-check others... A lot of these keywords seem to be shortcuts for something(cvda) and produce errors when duplicated.